plugin/evm/mempool.go

```diff
@@ -142,16 +147,20 @@ func (m *Mempool) Add(tx *GossipAtomicTx) error {
 	m.ctx.Lock.RLock()
 	defer m.ctx.Lock.RUnlock()
 
-	return m.AddTx(tx.Tx)
+	return m.addTx(tx.Tx, false)
```

Comment:
I think we should be marking the tx as discarded if it returns an error (other than `errTxAlreadyKnown`) right?